polkadot/node/core/pvf/prepare-worker/src/lib.rs

```diff
@@ -328,3 +215,176 @@ fn runtime_construction_check(
 		.map(|_runtime| ())
 		.map_err(|err| PrepareError::RuntimeConstruction(format!("{:?}", err)))
 }
+
+#[derive(Encode, Decode)]
+struct Response {
+	artifact: CompiledArtifact,
+	memory_stats: MemoryStats,
+}
+
+async fn handle_child_process(
+	pvf: PvfPrepData,
+	mut pipe_write: os_pipe::PipeWriter,
+	preparation_timeout: Duration,
+	prepare_job_kind: PrepareJobKind,
+	executor_params: Arc<ExecutorParams>,
+) -> ! {
+	nix::sys::resource::setrlimit(
+		Resource::RLIMIT_CPU,
+		preparation_timeout.as_secs(),
+		preparation_timeout.as_secs(),
+	)
+	.unwrap_or_else(|_| process::exit(libc::EXIT_FAILURE));
```

Comment:
Can we write the error to the pipe before exiting? (We can probably have a helper function that writes to the pipe and exits.)